File: src/erlang/ch01/1.2/l0/src/exercises_starter.erl
-module(exercises_starter).
-export([exercise1/0, exercise2/0, exercise3/0, exercise4/0, main/0]).

exercise1() -> 326 + 589.
exercise2() -> 704 - 259.
exercise3() -> 38 * 12.
exercise4() -> {123 div 7, 123 rem 7}.

main() ->
    io:format("Exercise 1 (326+589): ~p~n", [exercise1()]),
    io:format("Exercise 2 (704-259): ~p~n", [exercise2()]),
    io:format("Exercise 3 (38*12): ~p~n", [exercise3()]),
    {Q, R} = exercise4(),
    io:format("Exercise 4 (123//7, 123%7): ~p ~p~n", [Q, R]).

File: archived/src/python/ch01/1.1/l1/exercises_starter.py
from twos_complement import to_twos_complement, from_twos_complement

def exercise_1():
    # Convert decimal 100 to binary, octal, hex
    n = 100
    return bin(n), oct(n), hex(n)

def exercise_2():
    # Write -7 in 8-bit two's complement
    return to_twos_complement(-7, 8)

def exercise_3():
    # Verify 0xFF == 255
    return (0xFF == 255)

def exercise_4():
    # Parse "11111001" as 8-bit two's complement
    return from_twos_complement("11111001")

def _demo():
    print("Exercise 1 (100):", exercise_1())        # ('0b1100100', '0o144', '0x64')
    print("Exercise 2 (-7):", exercise_2())        # '11111001'
    print("Exercise 3 (0xFF):", exercise_3())      # True
    print("Exercise 4 (11111001):", exercise_4())  # -7

if __name__ == "__main__":
    _demo()
